crates/admin/src/cluster_controller/logs_controller.rs

```diff
@@ -422,6 +435,26 @@ impl LogletConfiguration {
             }
         }
     }
+
+    fn node_set_iter(&self) -> impl Iterator<Item = &PlainNodeId> {
+        match self {
+            LogletConfiguration::Replicated(configuration) => {
+                itertools::Either::Left(configuration.nodeset.iter())
+            }
+            LogletConfiguration::Local(_) => itertools::Either::Right(iter::empty()),
+            LogletConfiguration::Memory(_) => itertools::Either::Right(iter::empty()),
+        }
```

Comment:
I hadn't encountered this trick before, nice! Previously I've resorted to dynamic dispatch in situations where I needed to return one of several different kinds of iterators.

Reply:
In the end, it might not be all that different 😅